gm4_mountaineering/data/gm4_mountaineering/functions/ski_effects/check_snowy_block.mcfunction

```diff
@@ -0,0 +1,15 @@
+# clear existing tags, check if the player is on a snowy block and wearing skis, add the skiing tag back if yes, and add ski_off_snow tag if not on snowy block
+# @s = player
+# at @s
+# run from main
+
+# Remove tags unless the player is in the air
+execute unless block ~ ~-0.1 ~ minecraft:air run tag @s remove gm4_mountaineering_skiing
+execute unless block ~ ~-0.1 ~ minecraft:air run tag @s remove gm4_mountaineering_skiing_off_snow
+
+# Add skiing tag if on snow layer or any other snowy block (contained within the predicate) with skis
+execute if block ~ ~ ~ minecraft:snow if predicate gm4_mountaineering:wearing_skis run tag @s add gm4_mountaineering_skiing
+execute if predicate gm4_mountaineering:on_snowy_block if predicate gm4_mountaineering:wearing_skis run tag @s add gm4_mountaineering_skiing
```

Comment:
You should be able to combine these two by adding a check for the snow block in your `on_snowy_block` predicate

Reply:
If I recall, the reason I had to split this up is because the predicate couldn't detect `snow`. (It detects `snow block` fine, but for some reason `snow`, referring to just the single layer, had to be called separately as done in this function)